usergetterproxy.go

```diff
@@ -0,0 +1,51 @@
+package security
+
+import "net/http"
+
+// UserGetterProxy switches between UserGetters depending on issuer/clientid of the token in the request.
+type UserGetterProxy struct {
+	ugs       map[string]UserGetter
+	defaultUG UserGetter
+}
+
+// NewUserGetterProxy creates a new UserGetterProxy with the given default UserGetter which is
+// used if no other match is found.
+func NewUserGetterProxy(defaultUG UserGetter) *UserGetterProxy {
+	return &UserGetterProxy{
+		ugs:       make(map[string]UserGetter),
+		defaultUG: defaultUG,
+	}
+}
+
+// Add adds the given UserGetter for the specified issuer/clientid combination that takes precedence
+// over the default UserGetter if matched.
+func (u *UserGetterProxy) Add(issuer, clientid string, ug UserGetter) {
```

Comment:
It is pendantic, i know, but this only works, if the `Add` function is called when initializing the `UserGetterProxy`. The signature `Add` sounds like it can be done at runtime too, but in this case you should use a `RWMutex` to protected the map.

So i think it would be cleaner to use a lock here OR to move the adding of `UserGetter`s to the constructor funtion with option-parameters, so after construction the private fields are immutable

Reply:
Not pedantic at all, in my optinion. Implemented as option param - should be used at initialization time only

We should consider to put all of the components in different packages so that the names can be more concise.